stan/math/rev/fun/quad_form.hpp

```diff
@@ -93,6 +93,65 @@ class quad_form_vari : public vari {
 
   quad_form_vari_alloc<Ta, Ra, Ca, Tb, Rb, Cb>* impl_;
 };
+
+/**
+ * Return the quadratic form \f$ B^T A B \f$.
+ *
+ * Symmetry of the resulting matrix is not guaranteed due to numerical
+ * precision.
+ *
+ * @tparam T1 type of the first (square) matrix
+ * @tparam T2 type of the second matrix
+ *
+ * @param A square matrix
+ * @param B second matrix
+ * @return The quadratic form, which is a symmetric matrix.
+ * @throws std::invalid_argument if A is not square, or if A cannot be
+ * multiplied by B
+ */
+template <typename Mat1, typename Mat2,
+          require_all_matrix_t<Mat1, Mat2>* = nullptr,
+          require_any_var_matrix_t<Mat1, Mat2>* = nullptr>
+inline auto quad_form_impl(const Mat1& A, const Mat2& B, bool symmetric) {
+  check_square("quad_form", "A", A);
+  check_multiplicable("quad_form", "A", A, "B", B);
+
+  using return_t
+    = promote_var_matrix_t<decltype(value_of(B).transpose().eval() *
+				    value_of(A) *
+				    value_of(B).eval()), Mat1, Mat2>;
+
+  auto arena_A = to_arena(A);
+  auto arena_B = to_arena(B);
+
+  check_not_nan("multiply", "A", value_of(arena_A));
+  check_not_nan("multiply", "B", value_of(arena_B));
+
+  auto arena_res = to_arena(value_of(arena_B).transpose() *
+			    value_of(arena_A) *
+			    value_of(arena_B));
+
+  if(symmetric) {
+    arena_res = (arena_res + arena_res.transpose()).eval();
+  }
+  
+  return_t res = arena_res;
```

Comment:
`arena_res` is on the arena, so since `return_t` is a `var<mat>` it will just take the pointer `arena_res` has right?

Reply:
Yes. `arena_t<var<T>>` is `var<T>`.